Use external documentation when it makes sense.

gperetin opened the pull request:
This is what it looks like in the UI:
![image](https://github.com/Pythagora-io/gpt-pilot/assets/192281/23fb826d-d1ca-43d9-b706-33ab55ab6b9f)

Also tested manually and confirmed that the doc snippets are in the `tasks` column in the DB.

senko commented on core/agents/external_docs.py:
```diff
@@ -0,0 +1,153 @@
+        for docset_key, snip in snippets:
+            docs.append({"key": docset_key, "desc": docsets_dict[docset_key], "snippets": snip})
+
+        self.next_state.current_task["docs"] = docs
```
I'd prefer the docs are stored directly in the project state, similar to how we handle relevant files and modified files.

The "tasks" structure is intended to capture the information for all the tasks that we need in each state, not as a free-form bag in which to put data relevant to the current state or current task. While one can argue that task instructions also fail this, we should not dig ourselves deeper in the hole by adding more stuff.

I know this adds a couple of extra minutes of work now, but if we do this we're never going to go back on it and it's going to make each new addition like this less objectionable,...

gperetin commented on core/agents/external_docs.py:
```diff
@@ -0,0 +1,153 @@
+
+        available_docsets = await self._get_available_docsets()
+        selected_docsets = await self._select_docsets(available_docsets)
+        telemetry.set("docsets_used", selected_docsets)
```
I thought we also do it at the end of each task, in `StateManager.log_task_completer` which `TaskCompleter` calls.
I'll switch to `trace_code_event` as suggested, both solutions should be once per task so no additional requests.

gperetin commented:
Addressed all the review comments, this is ready for review again.

This is what the output currently looks like:
![image](https://github.com/Pythagora-io/gpt-pilot/assets/192281/ed173924-fc55-4742-bb70-f2381aa3ae59)
